app/src/main/java/com/example/lesson05androidhwcalculator/HistoryAdapter.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.lesson05androidhwcalculator
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.lesson05androidhwcalculator.databinding.HistoryNotesBinding
+import java.util.ArrayList
+
+class HistoryAdapter(
+    private val historyList: ArrayList<String>?,
```

Comment:
везде нужно пользоваться интерфейсами, а не конкретными реализациями, у тебя адаптер ни к каким другим списком работать не будет (например LinkedList) + почему он nullable?
можно предоставлять дефолтную реализацию
```suggestion
    private val historyList: List<String> = emptyList(),
```